### DISCUSSION TITLE
Tracking the URL

### DISCUSSION THREAD

**Discussion by julianengel** [plausible/analytics#1649] [Anything Else]
Hi! I have a rather specific question for a particular use case. 

We have a template that we use for pages that aren't ready yet, but where the domain is registered. The code is the same, and the tracking link will be the same. Now the question is if there's a possibility to fetch the actual URL (the script grabs that as `a` I think anyways for checking query params, but without query params, there's no way to actually know which URL generated the page view. 

The question is: is there any way to see the referring URL in the dashboard that I missed? 

Thank you!

**Comment by metmarkosaric** [comment]
hi @julianengel! 

i'm not sure i completely understand your use case. we do have this way for you to track custom query parameters. see https://plausible.io/docs/custom-query-params. is this what you have in mind?

the referrals are in the Top Sources report but the actual referring URL is no longer sent by any of the major browsers. this is why it's no longer possible to see and you can only see the referring domain name and not the actual URL. this affects Plausible the same way as it affects GA and other analytics tools. you can read more here: https://plausible.io/blog/referrer-policy

**Comment by julianengel** [comment]
Hey @metmarkosaric!
Firstly, thank you for an amazing product and the fast reply! I am working on getting a sponsorship approved from my employer, you guys rock!

Sorry, it was a long day. 

Internally, when we register a domain but aren't yet ready to use it, we point it to our [launchpage.xyz ](https://launchpage.xyz)

We have a really simple tracker in, that just tells us how often the site of the waiting URL was visited. The other day someone asked if we could actually use something more advanced such as Plausible to actually get some stats and insight into our waiting domains and to gauge which project to tackle first. 

The use case we have: multiple domains serve the same landing page essentially. I was wondering if there's any way to add a UTM tag, or anything else to the URL before pinigin plausible, so that under sources it would show which domain the view was on (not from, as we're not redirecting).

Essentially, the improved question is: Is there any way to modify the URL passed to the plausible script? Without redirecting and reloading the page?

**Reply by metmarkosaric** [reply]
you're welcome and thanks for the kind words!

perhaps this custom locations feature does what you need? https://plausible.io/docs/custom-locations

**Reply by julianengel** [reply]
YES! That's exactly what I was looking for. Thank you so very much! This is perfect!

**Reply by metmarkosaric** [reply]
nice, thanks for the update! enjoy Plausible!

**Reply by julianengel** [reply]
I have a quick technical question, I did everything as per the code below, however the request still comes in as default page.     
I'm triggering the event as follows, but upon inspecting the request, it's just the `window.location.href `, and not the correct full updated one
`plausible('pageview', { u: window.location.href + '?utm_source=' + url });`

**Reply by RobertJoonas** [reply]
It seems like you're on self-hosted Plausible. This feature is not a part of self-hosted yet and will be included in the next release.

New features are released to the cloud version several times per week. Features are battle-tested in the cloud which allows us to fix any bugs before the general self-hosted release. Every six months we combine all the changes into a new self-hosted release.

**Reply by julianengel** [reply]
@RobertJoonas Thank you! Super sorry for the delay in getting back to you, this got flooded with my issues. I appreciate the help and insight into the matter, and wish you an absolutely awesome day!

**Reply by julianengel** [reply]
On a different note, if we move this domain from self-hosted to cloud, can we migrate the data with us?

**Reply by metmarkosaric** [reply]
it's unfortunately not possible to migrate the stats from self-hosted

**Reply by julianengel** [reply]
Ah, thank you... I see. That's a shame, but completely understandable considering how difficult that implementation could be.  Do you by any chance have an ETA on when the functionality for manual will be shipped with the self hosted version?

**Reply by metmarkosaric** [reply]
no eta but the last release was last month so the next one should be at some point this summer

**Reply by julianengel** [reply]
Thank you! Would you have any suggestion for a workaround in the meantime to be able to see the domain's actual URL? If not, completely cool, it's a longshot :)

**Reply by metmarkosaric** [reply]
it's not possible on self hosted unfortunately. it's a brand new feature on Plausible and should make it to self hosted on the next release

**Comment by julianengel** [comment]
Hi @metmarkosaric, any update on when the next release will go out? Thank you!

**Reply by ruslandoga** [reply]
@julianengel 👋

Do you mean a self-hosted one? If so, you might be interested in https://github.com/plausible/analytics/discussions/2324

**Reply by julianengel** [reply]
YESS!! Thank you! The manual is the GOAT! We can retire our hacked together solution :)
